src/Features/Core/Portable/CodeFixesAndRefactorings/CodeActionRequestPriorityProvider.cs

```diff
@@ -81,25 +87,33 @@ public static bool MatchesPriority(this ICodeActionRequestPriorityProvider provi
             return true;
         }
 
-        if (provider.Priority == CodeActionRequestPriority.Low)
+        if (provider.Priority == codeFixProvider.RequestPriority)
+        {
+            return true;
+        }
+
+        if (provider.Priority == CodeActionRequestPriority.Low && provider.HasDeprioritizedAnalyzerSupportingDiagnosticId(codeFixProvider.FixableDiagnosticIds))
         {
             // 'Low' priority can be used for two types of code fixers:
             //  1. Those which explicitly set their 'RequestPriority' to 'Low' and
             //  2. Those which can fix diagnostics for expensive analyzers which were de-prioritized
             //     to 'Low' priority bucket to improve lightbulb population performance.
-            // Hence, when processing the 'Low' Priority bucket, we accept fixers with any RequestPriority,
-            // as long as they can fix a diagnostic from an analyzer that was executed in the 'Low' bucket.
+            // Hence, when processing the 'Low' Priority bucket and the priority provider indicates
+            // there was a de-prioritized analyzer supporting one of our fixable diagnostic ids, we accept
+            // fixers with any RequestPriority, as long as they can fix a diagnostic from an analyzer that
+            // was executed in the 'Low' bucket.
             return true;
         }
 
-        return provider.Priority == codeFixProvider.RequestPriority;
+        return false;
     }
 }
 
 internal sealed class DefaultCodeActionRequestPriorityProvider(CodeActionRequestPriority? priority = null) : ICodeActionRequestPriorityProvider
```

Comment:
i hate this optional value here.

Reply:
Would prefer to not change as part of this PR. Let me know if you feel strongly otherwise.

Reply:
It was a general gripe. You don't have to change it at all.